sysroot/usr/include/kernel/lib/stdio.hpp

```diff
@@ -11,11 +11,16 @@
 
 #include <stdint.h>
 #include <stdarg.h>
+#include <lib/mutex.hpp>
 
 #ifndef EOF
 #define EOF (-1)
 #endif
 
+typedef int (*fnptr_t)(unsigned c, void** helper);
```

Comment:
Shouldn't this be a `size_t` instead of an `int`?

Reply:
I just copied this from `stdio.cpp`, but I do believe `int` is correct. Why do you think it should be a `size_t`?

Reply:
If it's a pointer then it should be the length of whatever `size_t` is, right?

Reply:
`int` is the return value of the function pointer.

Reply:
Ah, right. That makes sense. Is `fnptr_t` actually used anywhere? If not, maybe we should just remove it?

Reply:
It's used in `px_do_printf` on the line below.

Reply:
Alright then. I'm not a fan of typedef'ing it, but we'll leave it for now.